engine/runtime/src/main/java/org/enso/interpreter/runtime/data/hash/EnsoHashMapBuilder.java

```diff
@@ -82,12 +88,57 @@ public StorageEntry[] getEntries(int atGeneration, int size) {
     }
   }
 
+  record Entry(Object key, Object value) {}
+
+  Iterator<Entry> getEntriesIterator(int atGeneration) {
+    return new EntriesIterator(atGeneration);
+  }
+
+  class EntriesIterator implements Iterator<Entry> {
```

Comment:
Thanks for removing any signs of `public` in this and `EnsoHashMap` files. Possibly you can also make this class `final`.

Reply:
Made final in a44cf5af6aab98c5d65ea7269d798f86acff64fd